hub/core/storage/provider.py

```diff
@@ -1,6 +1,8 @@
 from collections.abc import MutableMapping
 from typing import Optional
 from abc import ABC, abstractmethod
+from hub.util.assert_byte_indexes import assert_byte_indexes
+from hub.constants import BYTE_PADDING
 
 
 class Provider(ABC, MutableMapping):
```

Comment:
we should consider renaming `Provider` -> `StorageProvider` that way when we develop `hub.core.compute.provider`, it can be named `ComputeProvider` (instead of `Provider`, which would conflict with `hub.core.storage.provider`).

we should keep the `hub.core.[storage/compute].provider` file structure though, just name the classes `StorageProvider`/`ComputeProvider`.